### PR TITLE
support php8.2

### PR DESCRIPTION
**Pull request by kilvn**
## Types of Changes

- [] Support php 8.2.

## Definition of Done Checklist:

- [] I've read the [Contribution Guide](http://laradock.io/contributing).
- [] I've updated the **documentation**. (refer to [this](http://laradock.io/contributing/#update-the-documentation-site) for how to do so).
- [] I enjoyed my time contributing and making developer's life easier :)

### CODE REVIEW

File: workspace/Dockerfile [hunk 1]
```diff
@@ -334,7 +334,11 @@ RUN if [ ${INSTALL_XDEBUG} = true ]; then \
   # https://xdebug.org/docs/compat
   apt-get install -yqq pkg-config php-xml php${LARADOCK_PHP_VERSION}-xml && \
   if [ $(php -r "echo PHP_MAJOR_VERSION;") = "8" ] || { [ $(php -r "echo PHP_MAJOR_VERSION;") = "7" ] && { [ $(php -r "echo PHP_MINOR_VERSION;") = "4" ] || [ $(php -r "echo PHP_MINOR_VERSION;") = "3" ] ;} ;}; then \
-    pecl install xdebug-3.1.4; \
+    if [${LARADOCK_PHP_VERSION} = "8.2"]; then \
```
Comment (bestlong) [inline workspace/Dockerfile:337]: syntax problem ?

File: php-fpm/Dockerfile [hunk 2]
```diff
@@ -218,7 +218,11 @@ RUN if [ ${INSTALL_XDEBUG} = true ]; then \
   # Install the xdebug extension
   # https://xdebug.org/docs/compat
   if [ $(php -r "echo PHP_MAJOR_VERSION;") = "8" ] || { [ $(php -r "echo PHP_MAJOR_VERSION;") = "7" ] && { [ $(php -r "echo PHP_MINOR_VERSION;") = "4" ] || [ $(php -r "echo PHP_MINOR_VERSION;") = "3" ] ;} ;}; then \
-    pecl install xdebug-3.1.4; \
+    if [${LARADOCK_PHP_VERSION} = "8.2"]; then \
```
[the remark this answers is not in the document]
Comment (kilvn) [reply php-fpm/Dockerfile]: yes, fixed it.

File: workspace/Dockerfile (hunk 1 above)
Comment (kilvn) [reply workspace/Dockerfile]: yes, fixed it.

**Comment by kilvn** [comment]
I have pre checked in my own warehouse, and there is no problem with the changes to support PHP 8.2. However, due to the update of other software versions, there are some compilation errors unrelated to support PHP 8.2.

**Comment by parallels999** [comment]
```diff
Step 73/322 : RUN if [ ${INSTALL_XDEBUG} = true ]; then   apt-get install -yqq pkg-config php-xml php${LARADOCK_PHP_VERSION}-xml &&   if [ $(php -r "echo PHP_MAJOR_VERSION;") = "8" ] || { [ $(php -r "echo PHP_MAJOR_VERSION;") = "7" ] && { [ $(php -r "echo PHP_MINOR_VERSION;") = "4" ] || [ $(php -r "echo PHP_MINOR_VERSION;") = "3" ] ;} ;}; then     if [ ${LARADOCK_PHP_VERSION} = "8.2" ]; then       pecl install xdebug-3.2.0;     else       pecl install xdebug-3.1.4;     fi &&   else     if [ $(php -r "echo PHP_MAJOR_VERSION;") = "5" ]; then       pecl install xdebug-2.5.5;     else       if [ $(php -r "echo PHP_MINOR_VERSION;") = "0" ]; then         pecl install xdebug-2.9.0;       else         pecl install xdebug-2.9.8;       fi     fi   fi &&   echo "zend_extension=xdebug.so" >> /etc/php/${LARADOCK_PHP_VERSION}/cli/conf.d/20-xdebug.ini ;fi
 ---> Running in c7600da6fd31
-/bin/sh: 1: Syntax error: "else" unexpected
The command '/bin/sh -c if [ ${INSTALL_XDEBUG} = true ]; then   apt-get install -yqq pkg-config php-xml php${LARADOCK_PHP_VERSION}-xml &&   if [ $(php -r "echo PHP_MAJOR_VERSION;") = "8" ] || { [ $(php -r "echo PHP_MAJOR_VERSION;") = "7" ] && { [ $(php -r "echo PHP_MINOR_VERSION;") = "4" ] || [ $(php -r "echo PHP_MINOR_VERSION;") = "3" ] ;} ;}; then     if [ ${LARADOCK_PHP_VERSION} = "8.2" ]; then       pecl install xdebug-3.2.0;     else       pecl install xdebug-3.1.4;     fi &&   else     if [ $(php -r "echo PHP_MAJOR_VERSION;") = "5" ]; then       pecl install xdebug-2.5.5;     else       if [ $(php -r "echo PHP_MINOR_VERSION;") = "0" ]; then         pecl install xdebug-2.9.0;       else         pecl install xdebug-2.9.8;       fi     fi   fi &&   echo "zend_extension=xdebug.so" >> /etc/php/${LARADOCK_PHP_VERSION}/cli/conf.d/20-----xdebug.ini ;fi' returned a non-zero code: 2
-Service 'workspace' failed to build : Build failed
```
This is related to added xdebug conditional

**Comment by parallels999** [comment]
if you fix the tests that would be great

**Comment by kilvn** [comment]
> if you fix the tests that would be great

关于支持php8.2的测试已经通过，但是其他软件版本变化导致的编译报错暂时没时间研究

**Comment by andrekutianski** [comment]
workspace is building with PHP 8.2 and PHP 8.1, having `php` command mapped to 8.2 =/

**Pull request merged**